Samples/ios/SampleApp/SampleAppiOS/SamplePhotoUtil.m

```diff
@@ -0,0 +1,293 @@
+     {
+         if (error)
+         {
+             // What an app does on an InteractionRequired error will vary from app to app. More complex apps
+             // will usually want to present a notification to the user in an unobtrusive way (such as on a
+             // status bar in the application UI). Simpler apps, like this one, can jump straight into the
+             // acquireTokenInteractive flow.
+             if ([error.domain isEqualToString:MSALErrorDomain] && error.code == MSALErrorInteractionRequired)
+             {
+                 [SampleMSALUtil acquireTokenInteractiveForCurrentUser:^(NSString *token, NSError *error)
+                  {
+                      if (error)
+                      {
+                          photoBlock(nil, error);
+                          return;
+                      }
+                      
+                      [self getPhoto:token block:photoBlock];
+                  }];
+             }
+             else
+             {
+                 photoBlock(nil, error);
+             }
+             return;
+         }
+         
+         [self getPhoto:token block:photoBlock];
+     }];
+}
+
+
+static NSString * const kLastPhotoCheck = @"last_photo_check";
+
++ (void)clearPhotoCache
+{
+    [[NSUserDefaults standardUserDefaults] removeObjectForKey:kLastPhotoCheck];
+    
+    NSError *error = nil;
+    if (![[NSFileManager defaultManager] removeItemAtPath:[self cachedImagePath] error:&error])
+    {
+        NSLog(@"Failed to remove cache file: %@", error);
+    }
+}
+
++ (void)setLastChecked
+{
+    [[NSUserDefaults standardUserDefaults] setObject:[NSDate date] forKey:kLastPhotoCheck];
+}
+
+#define SECONDS_PER_DAY 3600 * 24
+
++ (NSString *)cachedImageDirectory
+{
+    NSArray<NSString *> *directories = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);
+    return [NSString stringWithFormat:@"%@/com.microsoft.MSALSampleApp/userphoto", directories[0]];
+}
+
++ (NSString *)cachedImagePath
+{
+    return [NSString stringWithFormat:@"%@/%@", [self cachedImageDirectory], [SampleMSALUtil currentUserIdentifer]];
+}
+
++ (UIImage *)checkPhotoCache
+{
+    NSString *cachedImagePath = [self cachedImagePath];
+    if (!cachedImagePath)
+    {
+        return nil;
+    }
+    
+    UIImage *cachedImage = [UIImage imageWithContentsOfFile:cachedImagePath];
+    NSDate *lastChecked = [[NSUserDefaults standardUserDefaults] objectForKey:kLastPhotoCheck];
+    if (!cachedImage)
+    {
+        // If we never had an image don't check more then once a day...?
+        if (lastChecked && [lastChecked timeIntervalSinceNow] < SECONDS_PER_DAY)
+        {
+            return [UIImage imageNamed:@"no_photo"];
+        }
+    }
+    else if ([lastChecked timeIntervalSinceNow] < SECONDS_PER_DAY * 7)
+    {
+        // If we've checked in less then 7 days just return the cached image
+        return cachedImage;
```

Comment:
a little confused about the logic here. you're returning the cachedImage out of both the else if clause and outside it. What's the point?